packages/apps/stock/src/utils/line-navigation.ts

```diff
@@ -51,10 +53,17 @@ export const showLine = ({
 
   const {trackingNumberConfiguration} = product;
 
-  if (item?.data?.statusSelect === detailStatus) {
+  if (
+    item?.data?.statusSelect === detailStatus ||
+    (!skipTrackingNumberVerification &&
+      trackingNumberConfiguration &&
+      trackingNumber == null)
+  ) {
     navigation.navigate(lineDetailsScreen, {
       [itemLine.name]: updatedItemLine,
       [item.name]: item.data,
+      [`${itemLine.name}Id`]: updatedItemLine.id,
+      [`${productKey}Id`]: product.id,
```

Comment:
This will not work for supplier arrival as line details screen is not asking for id but complete item, I think the simplier solution is to modify supplier arrival line details screen by asking for id and fetching line